src/Libraries/CoreNodeModels/Selection.cs

```diff
@@ -17,12 +19,14 @@ namespace CoreNodeModels
     /// </summary>
     /// <typeparam name="TSelection">The type which is used to constrain the selection.</typeparam>
     /// <typeparam name="TResult">The type which is returned from the selection or subselection.</typeparam>
+    [DataContract]
     public abstract class SelectionBase<TSelection, TResult> : NodeModel
```

Comment:
This is the only way I have been able to prevent the freezing from occurring, even this `[JsonIgnore]` on all public properties.  The `[DataContract]` prevents serialization unless explicitly defined

Reply:
Not sure if this is the issue or not, but just in case it helps 😄  : where the deserialization is happening, have you tried specifying a [JsonSerializerSettings](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_JsonSerializerSettings.htm) settings object. This is where you declare how to handle `null`s or missing properties during deserialization.
I've found explicitly declaring how to handle these clears up this kind of issues.